### ISSUE TITLE
When remove an item the shipping is not updated correctly

### ISSUE THREAD

**Issue by giosueDelgado** [vendure-ecommerce/vendure#2540]
**Describe the bug**
When I use the removeItemFromOrder the shipping is not updated correctly.

The problem is that the applyPriceAdjustments inside the orderservice not take care of the configuration of "await this.orderCalculator.applyPriceAdjustments" where there is a option that ask for recalculate the shipping.

**To Reproduce**
1. Add one item with one shipping method
2. Add another item with another shippin method
3. Remove the first item
4. The shipping is wrong without call the setshippingmethod (force the recalculation)

**Expected behavior**
Add the option also in the services of orderservice (removeItemFromOrder, addOrderItem, ecc..) all service that inside use the applyPriceAdjustments to give the possibility to recalculate the shipping.

**Environment (please complete the following information):**
 - @vendure/core version: 2.1.1
 - Nodejs version 16

**Comment by michaelbromley** [comment]
Hi,

Can you test this against the latest version, v2.1.3? That release contains this fix which may be related:

- https://github.com/vendure-ecommerce/vendure/pull/2512

**Comment by giosueDelgado** [comment]
Sure I can test but I see in the code also in the latest version that the flag to recalculate the shipping is never set to true so I don't think it resolve. The issue in my opinion is that when for example I call the removeOrderLine in my service the output of that is a order updated also for shipping if no-one line is using this shippingline.

**Comment by giosueDelgado** [comment]
And this can be done, in my opinion, if in the applyPriceAdjustments there is also the update of the shipping price based on the strategy of the shippingorderlinestrategy what do you think?

**Comment by michaelbromley** [comment]
The "recalculateShipping" flag effectively defaults to "true": https://github.com/vendure-ecommerce/vendure/blob/54953c5173f26becdf101f580958b9897ac81a12/packages/core/src/service/helpers/order-calculator/order-calculator.ts#L100-L103

I'm struggling to clearly understand the reproduction. Can you provide exact steps including the exact sequence of queries & mutations required to reproduce?

**Comment by giosueDelgado** [comment]
Ok, I checked deeper the problem.

I was thinking that "when I add an item to the cart (calling addOrderItem for example) in the service core  also the logic to associate the line to the shipping line was running" but in reality, this is done only in the setShippingMethod of orderService. 

In our use case when a user add an item it set also the type of shipping and we set using the customfield on order line.

So I don't know if is correct what I thinking or not let me know if I was clear.
